server/src/main/java/org/opensearch/index/translog/transfer/TransferService.java

```diff
@@ -66,6 +66,23 @@ void uploadBlobs(
      */
     void uploadBlob(final TransferFileSnapshot fileSnapshot, Iterable<String> remotePath, WritePriority writePriority) throws IOException;
 
+    /**
+     * Reads the input stream and uploads as a blob
+     * @param inputStream the stream to read from
+     * @param remotePath the remote path where upload should be made
+     * @param blobName the name of blob file
+     * @param writePriority Priority by which content needs to be written.
+     * @param listener the callback to be invoked once uploads complete successfully/fail
+     * @throws IOException the exception thrown while uploading
+     */
+    void uploadBlobAsync(
+        InputStream inputStream,
+        Iterable<String> remotePath,
+        String blobName,
```

Comment:
is blobName same as file name, can it be called as fileName to be in sync with other methods?